synapse_antispam/setup.py

```diff
@@ -7,5 +7,7 @@
     description="Mjolnir Antispam",
     include_package_data=True,
     zip_safe=True,
-    install_requires=[],
+    install_requires=[
+        "matrix-common >= 1.0.0"
```

Comment:
I don't think so; that would probably conflict with Synapse.
Maybe it'd make sense to say `< 2`?

Reply:
If we can hold ourselves to semver then `>= 1.0.0 < 2` seems like the right thing to do.
I can't see us easily implementing a sudden jump to `2.x` since we'd have to change Synapse & any dependent modules in lockstep though.

To avoid having to upgrade in lockstep, we'd have to phase breaking changes in by first being backward compatible in matrix-common, then gradually updating dependents, then finishing the breaking change. Maybe I'm just not thinking straight, but I don't know if semver expresses that?

Reply:
Synapse pins it FWIW.

Reply:
That kind of makes sense given the context at the time.

However we know it to be true that Mjolnir and one of its dependencies (Synapse) both depend on Matrix-Common. Having them both pin a specific version is asking for trouble where we can't upgrade the version in Synapse without cutting a new Mjolnir release with the new version pinned.

Reply:
Oh true! The web of dependencies is rather annoying!

Reply:
Yeah :/.

I've switched to `~= 1.0` as a result of David's suggestion and my reading of PEP 440.